sdk/core/azure-core/azure/core/pipeline/transport/_aiohttp.py

```diff
@@ -228,17 +228,25 @@ async def __anext__(self):
             except _ResponseStopIteration:
                 self.response.internal_response.close()
                 raise StopAsyncIteration()
-            except (ChunkedEncodingError, ConnectionError):
+            except (ChunkedEncodingError, ConnectionError) as ex:
                 retry_total -= 1
                 if retry_total <= 0:
                     retry_active = False
                 else:
                     await asyncio.sleep(retry_interval)
                     headers = {'range': 'bytes=' + str(self.downloaded) + '-'}
-                    resp = await self.pipeline.run(self.request, stream=True, headers=headers)
-                    if resp.http_response.status_code == 416:
-                        raise
-                    chunk = await self.response.internal_response.content.read(self.block_size)
+                    try:
+                        resp = await self.pipeline.run(self.request, stream=True, headers=headers)
+                        if not resp.http_response:
+                            raise
+                        if resp.http_response.status_code == 416:
```

Comment:
This will never be true, as `not resp.http_response` will always be fulfilled if status_code is 416